Documentation/ServicingGranularity.md

```diff
@@ -0,0 +1,192 @@
+# Servicing Granularity
+
+This doc is a more detailed description of the problems in [source-build#923
+"Figure out how source-build will work with CoreFX per-package servicing
+policy"](https://github.com/dotnet/source-build/issues/923).
+
+## "Build only if changed"
+
+This is a maintenance strategy where if a product hasn't changed, it isn't
+rebuilt and won't be rereleased under a new patch version.
+
+> A *product* for the sake of this doc is any released asset where an
+independent decision is made whether that specific asset will be patched and
+shipped.
+
+For projects that release prebuilt binaries such as NuGet packages, this
+strategy is intuitive and ordinary. For example, when a new .NET Core SDK patch
+is being built to take a networking stack fix, the Microsoft servicing workflow
+doesn't involve building and releasing new MSBuild `.nupkg` files to the NuGet
+Gallery. When the patched SDK is built, old MSBuild binaries are retrieved from
+the NuGet Gallery.
+
+Linux distributions tend to follow the same strategy, where the product is a
+distro package. To create a distro package release, all the source code making
+it up is rebuilt. A component that has source code living in an external distro
+package is not rebuilt.
```

Comment:
A single source package is providing the sources for multiple 'distro' packages.
This is also what we are doing with .NET Core.

It's not clear for me how this should work with granular builds.

Reply:
Is there a good name for the source package + all distro packages that come from it? This is what I'm referring to here--each new source package produces a set of distro packages that is "a product". All of the distro packages are built and released at the same time, effectively.

If source-build were granular, I believe we'd have many smaller source packages. (And fewer (maybe just one) distro package per source package.)

Reply:
I don't know the correct name, maybe @omajid has a suggestion.
I'm fine with calling it a product.

Earlier in the text it says:

> Linux distributions tend to follow the same strategy, where the product is a distro package.

This should then be:

> Linux distributions tend to follow the same strategy, where the product are all the packages built from a single source package.